e2e/smoke_test.go

```diff
@@ -114,6 +114,74 @@ var _ = Describe("Smoke End To End Tests - against OpenShift Cluster with Istio
 			})
 		})
 
+		Context("scenario-1-basic-deployment create/delete", func() {
+			var registry string
+			BeforeEach(func() {
+				scenario = "scenario-1"
+			})
+			JustBeforeEach(func() {
+				BuildTestServicePreparedImage(namespace)
+				registry = GetDockerRegistryInternal()
+			})
+
+			It("should watch for changes in ratings service and serve it", func() {
+				productPageURL := GetIstioIngressHostname() + "/test-service/productpage"
+
+				Eventually(AllPodsReady(namespace), 5*time.Minute, 5*time.Second).Should(BeTrue())
+
+				Eventually(call(productPageURL, map[string]string{
+					"Host": GetGatewayHost(namespace)}),
+					3*time.Minute, 1*time.Second).Should(ContainSubstring("ratings-v1"))
+
+				// switch to different namespace - so we also test -n parameter of $ ike
+				<-testshell.Execute("oc project default").Done()
+
+				// when we start ike to create
+				ikeWithCreate := testshell.ExecuteInDir(tmpDir, "ike", "create",
+					"--deployment", "ratings-v1",
+					"-n", namespace,
+					"--route", "header:x-test-suite=smoke",
+					"--image", registry+"/"+namespace+"/istio-workspace-test-prepared:latest",
+					"-s", "test-session",
+				)
+				Eventually(ikeWithCreate.Done(), 1*time.Minute).Should(BeClosed())
+
+				// ensure the new service is running
+				Eventually(AllPodsReady(namespace), 5*time.Minute, 5*time.Second).Should(BeTrue())
+
+				// check original response
+				Eventually(call(productPageURL, map[string]string{
+					"Host":         GetGatewayHost(namespace),
+					"x-test-suite": "smoke"}),
+					3*time.Minute, 1*time.Second).Should(ContainSubstring("prepared-image"))
+
+				// but also check if prod is intact
+				Eventually(call(productPageURL, map[string]string{}), 3*time.Minute, 1*time.Second).
+					ShouldNot(ContainSubstring("prepared-image"))
+
+				// when we start ike to delete
+				ikeWithDelete := testshell.ExecuteInDir(tmpDir, "ike", "delete",
+					"--deployment", "ratings-v1",
+					"-n", namespace,
+					"-s", "test-session",
+				)
+				Eventually(ikeWithDelete.Done(), 1*time.Minute).Should(BeClosed())
+
+				// check original response
+				Eventually(call(productPageURL, map[string]string{
+					"Host":         GetGatewayHost(namespace),
+					"x-test-suite": "smoke"}),
+					3*time.Minute, 1*time.Second).
+					ShouldNot(ContainSubstring("prepared-image"))
+
+				// but also check if prod is intact
+				Eventually(call(productPageURL, map[string]string{
+					"Host": GetGatewayHost(namespace)}),
+					3*time.Minute, 1*time.Second).ShouldNot(ContainSubstring("prepared-image"))
```

Comment:
Was just wondering if that is sufficient check. I mean all other responses would be green here, including "app not ready" etc. Shouldn't we test for something more specific in addition to this text?

Reply:
Should(And(ContainSubstring(""ratings-v1"), Not(ContainSubstring("prepared-image)) ?

Reply:
https://github.com/Maistra/istio-workspace/pull/252/commits/3e251f06399354186112f34971c2fbaf1d098255